pkg/auth/client/impersonate.go

```diff
@@ -48,3 +51,21 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+func buildImpersonatingConfig(user user.Info, impersonatingConfig *restclient.Config) *restclient.Config {
```

Comment:
Why a pointer?  It makes reasoning about whether you're wrapping the same transport deeper and deeper over time harder since I have to find the callers.